src/common/utils/date-time/format.ts

```diff
@@ -28,3 +29,113 @@ export const formatDateRange = ({
 
 export const transformDateToApiFormat = (formDate: string): string =>
   format(parse(formDate, dateFormFormat, new Date()), dateApiFormat);
+
+export const dropMilliseconds = (time: string): string => time.slice(0, -3);
+
+type WeekdayIndexToShortNameMappings = {
+  [language in Language]: {
+    [weekdayType in WeekdayTypes]: string;
+  };
+};
+
+const weekdays: WeekdayIndexToShortNameMappings = {
+  fi: {
+    1: 'ma',
+    2: 'ti',
+    3: 'ke',
+    4: 'to',
+    5: 'pe',
+    6: 'la',
+    7: 'su',
+  },
+  sv: {
+    1: 'Mån',
+    2: 'Tis',
+    3: 'Ons',
+    4: 'Tors',
+    5: 'Fre',
+    6: 'Lör',
+    7: 'Sön',
+  },
+  en: {
+    1: 'Mon.',
+    2: 'Tue.',
+    3: 'Wed.',
+    4: 'Thu.',
+    5: 'Fri.',
+    6: 'Sat.',
+    7: 'Sun.',
+  },
+};
+
+function getWeekdayShortNameByIndexAndLang({
+  weekdayIndex,
+  language,
+}: {
+  weekdayIndex: WeekdayTypes;
+  language: Language;
+}): string {
+  return weekdays[language][weekdayIndex];
+}
+
+type WeekdaySpan = {
+  startIndex: number;
+  lastInsertedIndex: number;
+  endIndex?: number;
+};
+
+export function createWeekdaysStringFromIndices(
```

Comment:
This is very likely not the most elegant way to solve the creation of the weekdays string :D. If you figure and elegant alternative I'll change this.